src/testing/tests.cpp

```diff
@@ -301,6 +304,14 @@ ::select_test(const AllParam *const parameters_input,
         if constexpr (dim==3 && nstate==dim+2)  return std::make_unique<EulerSplitEntropyCheck<dim, nstate>>(parameters_input, parameter_handler_input);
     } else if(test_type == Test_enum::khi_robustness) {
         if constexpr (dim==2 && nstate==dim+2)  return std::make_unique<KHIRobustness<dim, nstate>>(parameters_input, parameter_handler_input);
+    } else if (test_type == Test_enum::advection_limiter) {
+        if constexpr (dim == 1 && nstate == 1) return std::make_unique<AdvectionLimiter<dim, nstate>>(parameters_input);
+        if constexpr (dim == 2 && nstate == 1) return std::make_unique<AdvectionLimiter<dim, nstate>>(parameters_input);
+    } else if (test_type == Test_enum::burgers_limiter) {
+        if constexpr (dim == 1 && nstate == 1) return std::make_unique<BurgersLimiter<dim, nstate>>(parameters_input);
+        if constexpr (dim == 2 && nstate == 2) return std::make_unique<BurgersLimiter<dim, nstate>>(parameters_input);
```

Comment:
```suggestion
        if constexpr (dim == nstate && dim<3) return std::make_unique<BurgersLimiter<dim, nstate>>(parameters_input);
```

Reply:
can use this also for `AdvectionLimiter`

Reply:
Fixed the conditions for BurgersLimiter and AdvectionLimiter.